### ISSUE TITLE
Plugin.xml autoupdate  "since" and "until" cannot span major versions

### ISSUE THREAD

**Issue by deadok22** [JetBrains/gradle-intellij-plugin#66]
I'd like to be able to use the very same plugin build on multiple IntelliJ versions. 

For now I have to disable updateSinceUntilBuild and set these values manually. It would be cool if I could specify version range instead.

**Comment by winterDroid** [comment]
:+1:

**Comment by zolotov** [comment]
Hi Sergey!

This is already implemented in `0.0.39`, see `sameSinceUntilBuild` property for details.

**Comment by zolotov** [comment]
Sorry, misread the issue. Ok, will think about range customizing from gradle configuration

**Comment by deadok22** [comment]
Cool, thanks!

May be I should file a separate issue for that, but it'd also be really cool to have a target which builds a plugin against all supported major versions and runs tests.

**Comment by zolotov** [comment]
@deadok22 

> May be I should file a separate issue for that, but it'd also be really cool to have a target which builds a plugin against all supported major versions and runs tests.

I don't think that it's responsibility of gradle-intellij-plugin. For me it looks like a job for CI-tool which is able to run your plugin on particular revision against any intellij version you want. 

E.g. in `intellij-go` plugin we run `master` branch at TeamCity against release branch (for now – 143) and snapshot of release branch (143-SNAPSHOT). Before EAP the master branch also has been run on snapshot of developing branch (144-SNAPSHOT).

**Comment by zolotov** [comment]
@deadok22 btw, do you have any suggestions about possible syntax of since/until numbers customisation? Also, do you think that customisation of `since` build is really needed (because I don't)?

**Comment by deadok22** [comment]
Yes, setting 'since' build manually doesn't make sense - you have to build against older version if you want to be backward-compatible.

As to syntax, I'd make ```intellij.version``` and ```intellij.sameSinceUntilBuild``` obsolete, and add ```intellij.sinceBuild``` and ```intellij.untilBuild```. ```intellij.sinceBuild``` would be used as ```intellij.version```, ```intellij.untilBuild``` would default to *.9999.

**Comment by etanshaul** [comment]
I noticed the milestone for this doesn't have an attached date. Was this feature still in the works by any chance? @deadok22's last suggestion would be ideal.

**Comment by zolotov** [comment]
@etanshaul yes, it will be implemented eventually. But it's not the major one since it has workarounds

**Comment by zolotov** [comment]
Fixed in `0.2-SNAPSHOT`. See README to find out how to use snapshot version.

Customizing since/until builds can be done via `patchPluginXml` task:

```
task patchPluginXml {
  sinceBuild '163.0'
  untilBuild '164.*'
}
